app/scripts/models/reliers/oauth.js

```diff
@@ -140,10 +146,10 @@ define(function (require, exports, module) {
       self._importRequiredSearchParam('client_id', 'clientId');
       self._importRequiredSearchParam('scope', 'scope');
       self.importSearchParam('state');
-      self.importSearchParam('redirect_uri', 'redirectUri');
+      self._importValidateOptionalSearchParam('redirect_uri', 'redirectUri', Validate.isUriValid, OAuthErrors);
       self.importSearchParam('access_type', 'accessType');
-      self.importSearchParam('redirectTo');
-      self.importSearchParam('verification_redirect', 'verificationRedirect');
+      self._importValidateOptionalSearchParam('redirectTo', 'redirectTo', Validate.isUriValid, OAuthErrors);
+      self._importValidateOptionalSearchParam('verification_redirect', 'verificationRedirect', Validate.isUriValid, OAuthErrors);
       self.importBooleanSearchParam('keys');
```

Comment:
Huh, if `keys` is not `true` or `false`, then a 500 will occur. That doesn't seem right. We cause the same 400 behavior.